src/app/client/src/app/modules/course-batch/services/course-batch/course-batch.service.ts

```diff
@@ -0,0 +1,97 @@
+import { of as observableOf, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { Injectable, Input, EventEmitter } from '@angular/core';
+import { ConfigService, ServerResponse } from '@sunbird/shared';
+import { PlayerService } from '@sunbird/core';
+import { SearchParam, LearnerService, UserService, ContentService, SearchService } from '@sunbird/core';
+import * as _ from 'lodash';
+
+@Injectable()
+export class CourseBatchService {
+  private _enrollToBatchDetails: any;
+  private _updateBatchDetails: any;
+  public updateEvent = new EventEmitter();
+  private _enrolledBatchDetails: any;
+  private defaultUserList: any;
+  courseHierarchy: any;
+
+  constructor(public searchService: SearchService, public userService: UserService, public content: ContentService,
+  public configService: ConfigService,
+  public learnerService: LearnerService, private playerService: PlayerService) {
+  }
+
+  getCourseHierarchy(courseId) {
+    if (this.courseHierarchy && this.courseHierarchy.identifier === courseId) {
+      return observableOf(this.courseHierarchy);
+    } else {
+      return this.playerService.getCollectionHierarchy(courseId).pipe(map((response: ServerResponse) => {
+        this.courseHierarchy = response.result.content;
+        return response.result.content;
+      }));
+    }
+  }
+
+  getUserList(requestParam: SearchParam = {}): Observable<ServerResponse> {
```

Comment:
we should cache users list with query in service property as object with key as query and value is array of users